### PR TITLE
bpo-32215: Fix performance regression in sqlite3

### PR DESCRIPTION
**Pull request by berkerpeksag**
https://bugs.python.org/issue32215

### CODE REVIEW

File: Modules/_sqlite/statement.c [hunk 1]
```diff
@@ -85,10 +85,10 @@ int pysqlite_statement_create(pysqlite_Statement* self, pysqlite_Connection* con
                 continue;
         }
 
-        self->is_dml = (PyOS_strnicmp(p, "insert ", 7) == 0)
-                    || (PyOS_strnicmp(p, "update ", 7) == 0)
-                    || (PyOS_strnicmp(p, "delete ", 7) == 0)
-                    || (PyOS_strnicmp(p, "replace ", 8) == 0);
+        self->is_dml = (PyOS_strnicmp(p, "insert", 6) == 0)
```
Comment (serhiy-storchaka) [inline Modules/_sqlite/statement.c:88]: Are there commands that start with "update", "delete", etc? Can a user function that starts with such prefix (for example "replacewhitespaces()") be used in this context?

**Review by serhiy-storchaka — COMMENTED** [review]
Would be nice to add tests.

Is the tab character recognized as a whitespace by Sqlite?

File: Modules/_sqlite/statement.c (hunk 1 above)
Comment (berkerpeksag) [reply Modules/_sqlite/statement.c:88]: > Can a user function that starts with such prefix (for example "replacewhitespaces()") be used in this context?

Yes, they can. The original intent of the old version was to make that check more robust and avoid cases like you've mentioned. I don't think there is a way to achieve this without relying on SQLite APIs, but then relying on newer APIs usually results seeing "I can't compile the sqlite3 module on my system" issues on the tracker :)

**Comment by berkerpeksag** [comment]
> Would be nice to add tests.

To test that the performance regression is fixed or to test that something like a user defined function (e.g. `replacewhitespaces()`) doesn't cause any problems?

> Is the tab character recognized as a whitespace by Sqlite?

Looking at https://github.com/mackyle/sqlite/blob/master/src/tokenize.c it does.

**Comment by serhiy-storchaka** [comment]
First, test that `\t` and `\n` instead of space are accepted. Is there any effect besides the performance regression?

Testing a user function starting with one of these prefixes would be nice too.

**Comment by berkerpeksag** [comment]
I'm going to merge this without tests to get it in for 3.7.1 and open a separate pull request for tests.

**Pull request merged**